pkg/executor/fscache/poolcache.go

```diff
@@ -156,14 +168,55 @@ func (c *PoolCache) service() {
 				continue
 			}
 
-			// concurrency should not be set to zero and
-			//sum of specialization in progress and specialized pods should be less then req.concurrency
-			if req.concurrency > 0 && (specializationInProgress+len(funcSvcGroup.svcs)) >= req.concurrency {
-				resp.error = ferror.MakeError(ferror.ErrorTooManyRequests, fmt.Sprintf("function '%s' concurrency '%d' limit reached.", req.function, req.concurrency))
-			} else {
-				funcSvcGroup.svcWaiting++
-				resp.error = ferror.MakeError(ferror.ErrorNotFound, fmt.Sprintf("function '%s' all functions are busy", req.function))
+			numPodsSpecialized := len(funcSvcGroup.svcs)
+
+			// we're either not specializing a pod, or we are and none of them have room for another request
+			// check if we can specialize another pod
+			if req.concurrency > 0 && numPodsSpecialized+numPodsSpecializing >= req.concurrency {
+				// we can't specialize another pod because our specialized pods + specializing pods is >= this function's concurrency limit
+				// check if we have any pods that are already specialized
+				if numPodsSpecialized > 0 {
+					// we have specialized pod(s), just pick one at random and use it for this request
+					svc := randomSvc(funcSvcGroup.svcs)
+					otelUtils.LoggerWithTraceID(req.ctx, c.logger).
+						Info("max concurrency reached, sending request to random pod",
+							zap.String("function", req.function),
+							zap.String("address", svc.val.Address),
+							zap.Int("active_requests", svc.activeRequests),
+							zap.Int("svc_waiting", funcSvcGroup.svcWaiting),
+							zap.Int("queue_len", funcSvcGroup.queue.Len()),
+						)
```

Comment:
This log feels like it could be fairly noisy for our big apps. Is it intended to be temporary? Should we make it a debug log?